opensearch-operator/controllers/configurationController.go

```diff
@@ -0,0 +1,70 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	//v1 "k8s.io/client-go/applyconfigurations/core/v1"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	opsterv1 "opensearch.opster.io/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ConfigurationReconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	logr.Logger
+	Instance *opsterv1.OpenSearchCluster
+}
+
+func (r *ConfigurationReconciler) Reconcile(controllerContext *ControllerContext) (*opsterv1.ComponentStatus, error) {
+	if controllerContext.OpenSearchConfig == nil || len(controllerContext.OpenSearchConfig) == 0 {
+		return nil, nil
+	}
+	namespace := r.Instance.Spec.General.ClusterName
+	clusterName := r.Instance.Spec.General.ClusterName
+	configMapName := clusterName + "-config"
+
+	// Add some default config for the security plugin
+	controllerContext.OpenSearchConfig = append(controllerContext.OpenSearchConfig, "plugins.security.audit.type: internal_opensearch\n"+
```

Comment:
General question: Don't you prefer to have OpenSearchConfig be of type map: String -> String, so:
* You don't need to deal with new lines adding at each
* Key and value easily readable here
* Spot problems (adding same key by mistake)
?

Reply:
Good idea, didn't think of that. I've changed the implementation to use a map.